tests/conftest.py

```
@@ -528,6 +528,212 @@ def logger(request):
+_GPU_MARKERS_REQUIRING_SNAPSHOT = ("gpu_1", "gpu_2", "gpu_4", "gpu_8")
+
+
+def _read_visible_vram_mb_sum(pynvml) -> Optional[float]:
+    """Sum of `memoryInfo.used` across all NVML-visible GPUs, in MiB.
+
+    Returns None on any NVML error so callers can skip the snapshot rather
+    than corrupting a test result with a partial reading.
+    """
+    try:
+        count = pynvml.nvmlDeviceGetCount()
+    except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)